wire-ios-sync-engine/Source/Use cases/GetE2eIdentityCertificatesUseCase.swift

```diff
@@ -0,0 +1,104 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+import WireCoreCrypto
+import X509
+import SwiftASN1
+
+// sourcery: AutoMockable
+public protocol GetE2eIdentityCertificatesUseCaseProtocol {
+    func invoke(conversationId: Data,
+                clientIds: [MLSClientID]) async throws -> [E2eIdentityCertificate]
+}
+
+final public class GetE2eIdentityCertificatesUseCase: GetE2eIdentityCertificatesUseCaseProtocol {
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+
+    public init(coreCryptoProvider: CoreCryptoProviderProtocol) {
+        self.coreCryptoProvider = coreCryptoProvider
+    }
+
+    public func invoke(conversationId: Data,
+                       clientIds: [MLSClientID]) async throws -> [E2eIdentityCertificate] {
+        let coreCrypto = try await coreCryptoProvider.coreCrypto(requireMLS: true)
+        let clientIds = clientIds.compactMap({ $0.clientID.data(using: .utf8)})
+        let wireIdentities = try await getWireIdentity(coreCrypto: coreCrypto, conversationId: conversationId, clientIDs: clientIds)
+        if wireIdentities.isEmpty {
+            return Array(repeating: .notActivated, count: clientIds.count)
+        } else {
+            return try wireIdentities.map({
+                try $0.toE2eIdenityCertificate()
+            })
+        }
+    }
+
+    private func getWireIdentity(coreCrypto: SafeCoreCryptoProtocol,
+                                 conversationId: Data, clientIDs: [Data]) async throws -> [WireIdentity] {
+        return try await coreCrypto.perform {
+            return try await $0.getDeviceIdentities(
+                conversationId: conversationId,
+                deviceIds: clientIDs)
+        }
+    }
+}
+
+extension WireIdentity {
+
+    func toE2eIdenityCertificate() throws -> E2eIdentityCertificate {
+        let pemDocument = try PEMDocument(pemString: self.certificate)
+        let certificate = try Certificate(pemDocument: pemDocument)
+        return E2eIdentityCertificate(
+            certificate: certificate,
+            certificateDetails: self.certificate,
+            certificateStatus: self.status.e2eIdentityStatus,
+            mlsThumbprint: self.thumbprint
+        )
+    }
+}
+
+private extension DeviceStatus {
+
+    var e2eIdentityStatus: E2EIdentityCertificateStatus {
+        switch self {
+        case .valid:
+            return .valid
+        case .expired:
+            return .expired
+        case .revoked:
+            return .revoked
+        @unknown default:
+            preconditionFailure("Should have a valid status")
+        }
+    }
+
+}
+
+extension E2eIdentityCertificate {
+    static var notActivated: E2eIdentityCertificate {
+        E2eIdentityCertificate(
+            certificateDetails: "",
```

Comment:
Are all the other properties irrelevant for the status `notActivated`?

Reply:
We don't use mls info from this instance. At this point we wouldn't have any info from core crypto